rbm_sample.py

```diff
@@ -85,30 +85,121 @@ def ising_magnetization(field):
     m = np.abs((field).mean(axis=1))
     return np.array([m, m * m])
 
-def ising_averages(mag_history, model_size, label=""):
-    # Bootstrap samples
-    resample_size = 50000
-    #get resample states
-    mag_resample = bootstrap_resample(mag_history[:, 0, :], n=resample_size)
-    
-    #Now take average across resampled states and std dev.
-    mag_avg = mag_resample.mean(axis=0)
-    mag_std = mag_resample.std(axis=0)
 
-    #now take mag_history[:, 0, :] (all states, m ,for all conc samples) and find susc, then input the susc to bootstrap to return n_resample * n_conc array of susceptibility
-    mag_err = model_size*((mag_history[:, 0, :]-mag_history[:, 0, :].mean(axis=0))**2)
-    susc_resample = bootstrap_resample(mag_err, n=resample_size)
-    #take average across resampled states and std dev.
-    susc_avg = susc_resample.mean(axis=0)
-    susc_std = susc_resample.std(axis=0)
 
-    #finally take average across concurrent samples, can also compare to adding std error in quad and divide by N_conc to account for taking mean
+###############################  WIP  ##########################################
+def ising_matrix():
+    N = parameters['ising']['size']
+    a = np.zeros(shape=(N*N,N*N))
+    for i in range (0,N):
+       for j in range (0,N):
+          a[N*i+j][N*i+(j+1)%N] -= 1
+          a[N*i+j][N*((i+1)%N)+j] -= 1
+    return torch.from_numpy(a)
+
+
+
+def ising_free_energy(v, ising_matrix, nstates, beta=1.0):
+    v = v.double()
+    ising_matrix_mult = torch.sum(torch.mul(v,(v.mm(ising_matrix))), 1)
+    return ising_matrix_mult.mul(-beta).exp()
+
```

Comment:
So I think possibly we should move this to the rbm_pytorch.py once we are confident it is working and remove the exp() so it is outputting the equivalent thing to free energy, then it should be more compatible with finding Z

Reply:
where to put ising_matrix is less obvious but I guess it is probably fine here for time being, since ising_free_energy just needs the hamiltonian as a tensor

Reply:
@wilsonmr it looks as if with some minor modifications it works well with the Z function. For now i've just redifined another Z just for ising but then it should be easy to define just one Z which works for both rbm and ising free energy. Yeah i think we should then move everythin in rbm_pytorch, even if all this is just ising model stuff, not strictly rbm

Reply:
yeah, it's unclear to me if it should go into rbm_pytorch because that does seem to be for generic rbm stuff. I pulled your code and it seems nice, would be interesting to see if the K-L goes to zero if we use all of this stuff combined